src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -110,4 +126,102 @@ private void addListingToBusinesses(PlacesSearchResult place) {
     allBusinesses.add(new Listing(name, formattedAddress, 
           placeLocation, rating, photos, types));
   }
+
+  public void checkNumberOfLocationsOfBusiness() {
+    GeoApiContext context = new GeoApiContext.Builder()
+            .apiKey(KEY)
+            .build();
+    try {
+      for(int i = 0; i<allBusinesses.size();i++){
+        currentBusiness = allBusinesses.get(i);
+        TextSearchRequest request = new TextSearchRequest(context)
+                                            .query(currentBusiness.getName())
+                                            .location(latLng)
+                                            .radius(50000);
+        similarBusinessesInTheArea = request.await();
+        if(similarBusinessesInTheArea.results.length > 1){
+          checkBusinessThroughLinkedin(currentBusiness.getName());
+        }
+      }
+
+    }catch(Exception e) {
+      LOGGER.warning(e.getMessage());
+    }  
+  }
+  
+  private void checkBusinessThroughLinkedin(String currentBusinessName) 
+                          throws GeneralSecurityException, IOException {
+    String cx = "001390425498086947771:mhzyevhmmxq"; //Your search engine
+    //Instance Customsearch
+    Customsearch cs = new Customsearch.Builder(
+                                GoogleNetHttpTransport.newTrustedTransport(), 
+                                JacksonFactory.getDefaultInstance(), 
+                                null) 
+                                .setApplicationName("linkedinSearch") 
+                                .setGoogleClientRequestInitializer(
+                                      new CustomsearchRequestInitializer(KEY)) 
+                                .build();
+    Customsearch.Cse.List list = cs.cse().list(currentBusinessName).setCx(cx); 
+    List<Result> searchJsonResults = list.execute().getItems();
+    String[] numberOfFollowers;
+    if (searchJsonResults!=null){
+      if(searchJsonResults.size() != 0){
+        Result linkedinBusiness = searchJsonResults.get(0);
```

Comment:
Is this for one business or all of them?

Reply:
One business only